validator/client/validator.go

```diff
@@ -1212,44 +1209,66 @@ func (v *validator) ChangeHost() {
 func (v *validator) filterAndCacheActiveKeys(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte, slot primitives.Slot) ([][fieldparams.BLSPubkeyLength]byte, error) {
 	ctx, span := trace.StartSpan(ctx, "validator.filterAndCacheActiveKeys")
 	defer span.End()
-
+	isEpochStart := slots.IsEpochStart(slot)
+	validatorStatusUnpopulated := false
 	filteredKeys := make([][fieldparams.BLSPubkeyLength]byte, 0)
 	statusRequestKeys := make([][]byte, 0)
+	validatorStatuses := make([]*validatorStatus, 0)
 	for _, k := range pubkeys {
-		_, ok := v.pubkeyToValidatorIndex[k]
-		// Get validator index from RPC server if not found.
+		sta, ok := v.pubkeyToStatus[k]
 		if !ok {
-			i, ok, err := v.validatorIndex(ctx, k)
-			if err != nil {
-				return nil, err
-			}
-			if !ok { // Nothing we can do if RPC server doesn't have validator index.
-				continue
-			}
-			v.pubkeyToValidatorIndex[k] = i
+			validatorStatusUnpopulated = true
+			// skip adding if not found
+			continue
 		}
 		copiedk := k
 		statusRequestKeys = append(statusRequestKeys, copiedk[:])
+		copiedStatus := sta
```

Comment:
Does this copy anything? isn´t  `sta` just a pointer?

Reply:
I should just remove this